hedera-node/hedera-app/src/main/java/com/hedera/node/app/records/impl/producers/StreamFileProducerConcurrent.java

```diff
@@ -106,18 +110,23 @@ public StreamFileProducerConcurrent(
     /** {@inheritDoc} */
     @Override
     public void initRunningHash(@NonNull final RunningHashes runningHashes) {
-        if (lastRecordHashingResult != null) {
-            throw new IllegalStateException("initRunningHash() can only be called once");
-        }
+        lock.lock(); // Block until the lock is acquired
+        try {
+            if (lastRecordHashingResult != null) {
+                throw new IllegalStateException("initRunningHash() can only be called once");
+            }
 
-        if (runningHashes.runningHash() == null) {
-            throw new IllegalArgumentException("The initial running hash cannot be null");
-        }
+            if (runningHashes.runningHash() == null) {
+                throw new IllegalArgumentException("The initial running hash cannot be null");
+            }
 
-        lastRecordHashingResult = completedFuture(runningHashes.runningHash());
-        lastRecordHashingResultNMinus1 = completedFuture(runningHashes.nMinus1RunningHash());
-        lastRecordHashingResultNMinus2 = completedFuture(runningHashes.nMinus2RunningHash());
-        lastRecordHashingResultNMinus3 = completedFuture(runningHashes.nMinus3RunningHash());
+            lastRecordHashingResult = completedFuture(runningHashes.runningHash());
```

Comment:
If we make some of these volatile, could we avoid any of the locks below (e.g. in `switchBlocks`)?